flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -206,6 +214,23 @@ void retriggerSubpartitionRequest(int subpartitionIndex) throws IOException {
 		return Optional.of(new BufferAndAvailability(next, moreAvailable, getSenderBacklog()));
 	}
 
+	@Override
+	public List<Buffer> getInflightBuffers(long checkpointId) throws IOException {
+		synchronized (receivedBuffers) {
+			final List<Buffer> inflightBuffers = new ArrayList<>(receivedBuffers.size());
+			for (Buffer buffer : receivedBuffers) {
+				if (!buffer.isBuffer() && parseCheckpointBarrierOrNull(buffer) != null) {
+					break;
+				}
+				inflightBuffers.add(buffer.retainBuffer());
+			}
+
+			triggeredCheckpointId = checkpointId;
```

Comment:
I think here it's impossible to get invalid `checkpointId` (less than triggered), but I'd add a check (`Precondition`).

Also, naming is a bit confusing:
1. triggered is ambiguous to me; `inProgressCheckpointId` ?
1. `getInflightBuffers` suggests it doesn't modify anything;  `requestInflightBuffers`

Reply:
Added the `checkState` and I agree that the naming is not ideal.

I renamed `getInflightBuffers` to `requestInflightBuffers` (as suggested), and used `lastRequestedCheckpointId` instead of `triggeredCheckpointId` to highlight its usage.